### PR TITLE
DEV-1313: Add additional map types

### PR DESCRIPTION
**Pull request by torbensky**
New Map Types:

- [x] add "hz_inner_wall" map
- [x] add "hz_columns" map
- [x] add "hz_rings" map
- [x] add "hz_rivers_bridges" map
- [x] add "hz_spiral" map
- [x] add "hz_scatter" map
- [x] add "hz_grow_box" map
- [x] add "hz_expand_box" map
- [x] add "hz_expand_scatter" map

### CODE REVIEW

File: maps/hazards.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+func (m RiverAndBridgesHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	if err := (StandardMap{}).SetupBoard(lastBoardState, settings, editor); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%dx%d", lastBoardState.Width, lastBoardState.Height)
+	hazards, ok := riversAndBridgesMaps[key]
+	if !ok {
+		return errors.New("Board size is not supported by this map")
```
Comment (torbensky) [inline maps/hazards.go:147]: should this error or just ignore and leave hazards empty?

File: rand.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import "math/rand"
 
 type Rand interface {
 	Intn(n int) int
+	Range(min, max int) int
```
Comment (robbles) [inline rand.go:7]: So far this interface has just been a subset of the methods of a `math/rand.Rand`. This new method does look pretty useful, but can you add a comment explaining what it does, boundary conditions, etc?

File: maps/hazards.go [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+		Author: "Battlesnake",
+	}
+}
+
+func (m RiverAndBridgesHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	if err := (StandardMap{}).SetupBoard(lastBoardState, settings, editor); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%dx%d", lastBoardState.Width, lastBoardState.Height)
```
Comment (robbles) [inline maps/hazards.go:544]: optional: Could probably use a `rules.Point` or equivalent as the map key here and avoid using fmt?

File: maps/helpers.go [hunk 4]
```diff
@@ -87,3 +91,91 @@ func (m StubMap) UpdateBoard(previousBoardState *rules.BoardState, settings rule
 	}
 	return nil
 }
+
+// drawRing draws a ring of hazard points offset from the outer edge of the board
+func drawRing(bw, bh, hOffset, vOffset int) ([]rules.Point, error) {
+	if bw < 1 {
+		return nil, errors.New("board width too small")
```
Comment (robbles) [inline maps/helpers.go:98]: Probably should be a `rules.RulesetError` here (and in the other checks too) so engine can detect an invalid configuration.

**Review by robbles — COMMENTED** [review]
Loking great! I only have a few suggestions.
I'll try to test these out E2E this morning.

File: maps/hazards.go (hunk 1 above)
Comment (torbensky) [reply maps/hazards.go]: done!

File: rand.go (hunk 2 above)
Comment (torbensky) [reply rand.go:9]: done!

Comment (torbensky) [reply rand.go:10]: @robbles do you feel that the comments adequately explain this now?

**Pull request merged**